core/src/main/java/org/infinispan/commands/functional/AbstractWriteKeyCommand.java

```diff
@@ -2,26 +2,41 @@
 
 import static org.infinispan.commons.util.Util.toStr;
 
+import org.infinispan.cache.impl.CacheEncoders;
+import org.infinispan.cache.impl.EncodingClasses;
 import org.infinispan.commands.CommandInvocationId;
 import org.infinispan.commands.write.AbstractDataWriteCommand;
 import org.infinispan.commands.write.ValueMatcher;
 import org.infinispan.commons.util.EnumUtil;
+import org.infinispan.factories.ComponentRegistry;
 import org.infinispan.functional.impl.Params;
 
 public abstract class AbstractWriteKeyCommand<K, V> extends AbstractDataWriteCommand implements FunctionalCommand<K, V> {
 
    Params params;
    ValueMatcher valueMatcher;
    boolean successful = true;
+   EncodingClasses encodingClasses;
+   CacheEncoders cacheEncoders = new CacheEncoders();
```

Comment:
empty encoders handle the code where no encoding exist and so Entry code is the same with and without encoding. This can be discussed from a performance/memory point of view and try to make it better !